src/index.js

```diff
@@ -59,14 +58,59 @@ function cleanupAtContainer(container) {
   mountedContainers.delete(container)
 }
 
+const originalChange = fireEvent.change
+fireEvent.change = function reactChange(node, init) {
+  if (init && init.target && init.target.hasOwnProperty('value')) {
+    setNativeValue(node, init.target.value)
+  }
+  return originalChange(node, init)
+}
+
+// probably there is a better way to get access to a component instance
+const [
+  getInstanceFromNode,
+] = ReactDOM.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.Events
+const createReactEventSimulator = eventName => (node, init) => {
+  const propName = `on${eventName.charAt(0).toUpperCase()}${eventName.slice(1)}`
+  const simulate = getInstanceFromNode(node).memoizedProps[propName]
+
+  if (init && init.target && init.target.hasOwnProperty('value')) {
+    setNativeValue(node, init.target.value)
+  }
+
+  return simulate({
+    target: node,
+    type: eventName,
+    ...init,
+  })
+}
+
+// function written after some investigation here:
+// https://github.com/facebook/react/issues/10135#issuecomment-401496776
+function setNativeValue(element, value) {
```

Comment:
Maybe we should export this function from `dom-testing-library` so we can reuse that code here 🤔 

Would you like to make that PR there first? Should be a pretty simple one :)

Reply:
Good idea! :)

Reply:
Added export in the `dom-testing-library`:
https://github.com/kentcdodds/dom-testing-library/pull/137